controllers/capabilities/tenant_internal_reconciler.go

```diff
@@ -41,53 +40,69 @@ func NewTenantInternalReconciler(k8sClient client.Client, tenantR *apiv1alpha1.T
 // - Have 3scale Tenant Account
 // - Have active admin user
 // - Have secret with tenant's access_token
-func (r *TenantInternalReconciler) Run() error {
-	tenantDef, err := r.reconcileTenant()
+func (r *TenantInternalReconciler) Run() (ctrl.Result, error) {
+	res, err := r.reconcileTenant()
 	if err != nil {
-		return err
+		return ctrl.Result{}, err
 	}
 
-	adminUserDef, err := r.reconcileAdminUser(tenantDef)
-	if err != nil {
-		return err
+	if res.Requeue {
+		return res, nil
 	}
 
-	err = r.reconcileAccessTokenSecret(tenantDef)
+	err = r.reconcileAdminUser()
 	if err != nil {
-		return err
+		return ctrl.Result{}, err
 	}
 
-	tenantStatus := r.getTenantStatus(tenantDef, adminUserDef)
-
-	return r.updateTenantStatus(tenantStatus)
+	return ctrl.Result{}, err
 }
 
 // This method makes sure that tenant exists, otherwise it will create one
 // On method completion:
 // * tenant will exist
 // * tenant's attributes will be updated if required
-func (r *TenantInternalReconciler) reconcileTenant() (*porta_client_pkg.Tenant, error) {
-	tenantDef, err := controllerhelper.FetchTenant(r.tenantR, r.portaClient)
+func (r *TenantInternalReconciler) reconcileTenant() (ctrl.Result, error) {
+	tenantDef, err := controllerhelper.FetchTenant(r.tenantR.Status.TenantId, r.portaClient)
 	if err != nil {
-		return nil, err
+		return ctrl.Result{}, err
 	}
 
 	if tenantDef == nil {
 		tenantDef, err = r.createTenant()
 		if err != nil {
-			return nil, err
+			return ctrl.Result{}, err
 		}
-	} else {
```

Comment:
> why are we triggering the syncTenant reconcile right after creation instead of leaving it as it was with "if not present -> create, otherwise sync? 

I tried to explain in the comment just following the `createTenant` call:
```
// Early save access token as it is only available on the response of the
		// tenant creation call
```

The access token is only available on a successful token creation call response. So, it needs to be persisted in the secret and the ID of the tenant in the status. Doing something else and getting an error would imply to create a new tenant in the next reconcile run